templates/postgresql.10.conf.j2

```diff
@@ -126,8 +126,22 @@ temp_buffers = 24MB			# min 800kB
 					# (change requires restart)
 # Caution: it is not advisable to set max_prepared_transactions nonzero unless
 # you actively intend to use prepared transactions.
-work_mem = 512MB				# min 64kB
-maintenance_work_mem = 1024MB		# min 1MB
+
+# https://info.crunchydata.com/blog/optimize-postgresql-server-performance
+# Set work_mem to 0.5% of the total available memory
+# it can be allocated multiple time concurrently)
+{% if postgres_work_mem is defined %}
+work_mem = {{ postgres_work_mem }}MB		# min 64kB
+{% else %}
+work_mem = {{ ansible_memtotal_mb * 0.005 }}MB		# min 64kB
```

Comment:
We do not enforce the minimum value but it corresponds to ansible_memtotal_mb = 12.8 MB.

I believe it's ok